コンパイル時に、
```
segment.cpp: In member function 'bool Segment::isAdjacentTo(int) const':
segment.cpp:55:10: warning: enumeration value 'INVALID' not handled in switch [-Wswitch]
   switch (type_) {
          ^
```
のwarningがでるようになっちゃったから、switchの最後に
```
default:
 assert(false);
```
みたいなの入れるなりでwarningの解消をお願いします。

--- a/main.cpp
+++ b/main.cpp
@@ -1,5 +1,6 @@
 #include <cassert>
 #include <iostream>
+#include <memory>
 #include <set>
 #include <string>
 
@@ -26,12 +27,16 @@ int main(int argc, char const* argv[])
   TileFactory f;
   f.loadResource("tiles.json");
   std::string file_name("doc/championship_japan_2015_final.json");
-  ScoreSheet* score_sheet = readScoreSheetFromFile(file_name);
-  bool is_valid = validateScoreSheet(f, *score_sheet);
-  if (is_valid) {
-    std::cout << "valid" << std::endl;
+  std::unique_ptr<ScoreSheet> score_sheet(readScoreSheetFromFile(file_name));
+  if (score_sheet.get() == nullptr) {
+    std::cout << "Failed to read score sheet file" << std::endl;
   } else {
-    std::cout << "invaid" << std::endl;
+    bool is_valid = validateScoreSheet(f, *score_sheet.get());
+    if (is_valid) {
+      std::cout << "valid" << std::endl;
+    } else {
+      std::cout << "invaid" << std::endl;
+    }
   }
 
   return 0;

--- a/score_sheet.cpp
+++ b/score_sheet.cpp
@@ -15,9 +15,9 @@
 
 using json = nlohmann::json;
 
-Placement* __constructPlacement(json& j, std::map<std::string, MeepleColor>& player_map);
-TilePlacement* __constructTilePlacement(json& j, PlacementType type);
-MeeplePlacement* __constructMeeplePlacement(json& j);
+Placement* __constructPlacement(const json& j, const std::map<std::string, MeepleColor>& player_map);
+TilePlacement* __constructTilePlacement(const json& j, PlacementType type);
+MeeplePlacement* __constructMeeplePlacement(const json& j);
 
 SegmentType stringToSegmentType(std::string& str) {
   if (str == "city") {
@@ -77,7 +77,7 @@ ScoreSheet* readScoreSheetFromFile(const std::string& file_name) {
   return score_sheet;
 }
 
-Placement* __constructPlacement(json& j, std::map<std::string, MeepleColor>& player_map) {
+Placement* __constructPlacement(const json& j, const std::map<std::string, MeepleColor>& player_map) {
   std::string type_name = j["type"];
   PlacementType type = stringToPlacementType(type_name);
   if (type == PlacementType::INVALID) {
@@ -89,10 +89,12 @@ Placement* __constructPlacement(json& j, std::map<std::string, MeepleColor>& pla
   } else {
     if (j.find("player") != j.end()) {
       std::string player_name = j["player"];
-      if (player_map.find(player_name) == player_map.end()) {
+      auto it = player_map.find(player_name);
+      if (it == player_map.end()) {
 	return nullptr;
+      } else {
+	player = it->second;
       }
-      player = player_map[player_name];
     } else {
 	return nullptr;
     }
@@ -113,7 +115,7 @@ Placement* __constructPlacement(json& j, std::map<std::string, MeepleColor>& pla
   return new Placement(type, player, tile_name, tile_placement, meeple_placement);
 }
 
-TilePlacement* __constructTilePlacement(json& j, PlacementType type) {
+TilePlacement* __constructTilePlacement(const json& j, PlacementType type) {
   if (j.find("tilePlacement") == j.end()) {
     if (type == PlacementType::INITIAL) {
       return new TilePlacement(0, 0, 0);
@@ -147,7 +149,7 @@ TilePlacement* __constructTilePlacement(json& j, PlacementType type) {
   return new TilePlacement(x, y, rotation);
 }
 
-MeeplePlacement* __constructMeeplePlacement(json& j) {
+MeeplePlacement* __constructMeeplePlacement(const json& j) {
   assert(j.find("meeplePlacement") != j.end());
   std::string segment_name = j["meeplePlacement"]["segmentType"];
   SegmentType type = stringToSegmentType(segment_name);

--- a/score_sheet.cpp
+++ b/score_sheet.cpp
@@ -10,6 +10,7 @@
 #include "meeple_color.hpp"
 #include "score_sheet.hpp"
 #include "segment.hpp"
+#include "tile.hpp"
 #include "tile_factory.hpp"
 
 using json = nlohmann::json;

--- a/score_sheet.hpp
+++ b/score_sheet.hpp
@@ -60,7 +60,7 @@ class Placement {
     }
     ~Placement() {
       delete tile_placement_;
-      if (meeple_placement_ == nullptr) {
+      if (meeple_placement_ != nullptr) {
 	delete meeple_placement_;
       }
     }

